api/client.go

```diff
@@ -68,14 +71,23 @@ func NewClient(cfg Config) (Client, error) {
 	}
 	u.Path = strings.TrimRight(u.Path, "/")
 
+	var h map[string]string
+	if cfg.Headers != nil {
+		h = cfg.Headers
+	} else {
+		h = map[string]string{}
+	}
```

Comment:
Iterating over a `nil` map (in line 112) is just fine, so you don't need this. Also see comment about line 83 below.